### PR TITLE
ZTE-SH-CN-cluster-administration-federation-2017-09-19

### PR DESCRIPTION
**Pull request by XuJun00192603**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.8 Features: set Milestone to `1.8` and Base Branch to `release-1.8`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5541)

### CODE REVIEW

File: cn/docs/concepts/cluster-administration/federation.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+* 高可用性：通过跨集群分摊负载，自动配置DNS服务和负载均衡，联邦将集群失败所带来的影响降到最低。
+* 避免供应商锁定：跨集群使迁移应用程序变得更容易，联邦服务避免了供应商锁定。
+
+
+只有在多个集群的场景下联邦服务才是有帮助的。这里列出了一些你会使用多个集群的原因：
+
+* 降低延迟：在多个区域含有集群，可使用离用户最近的集群来服务用户，从而最大限度降低延迟。
+* 故障隔离：对于故障隔离，也许有多个小的集群比有一个大的集群要更好一些（例如：一个云供应商的不同可用域里有多个集群）。详细信息请参阅[多集群指南](/docs/admin/multi-cluster)。
+* 可伸缩性：对于单个kubernetes集群是有伸缩性限制的（但对于大多数用户来说并非如此。更多细节参考[Kubernetes扩展和性能目标](https://git.k8s.io/community/sig-scalability/goals.md)）。
+* [混合云](#hybrid-cloud-capabilities)：可以有多个集群，它们分别拥有不同的云供应商或者本地数据中心。
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/concepts/cluster-administration/federation.md]: #hybrid-cloud-capabilities
对应你下文的
### 混合云的能力
所以这里应该是中文

File: cn/docs/concepts/cluster-administration/federation.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+* 增加网络的带宽和损耗：联邦控制平台会监控所有的集群，来确保集群的当前状态与预期一致。那么当这些集群运行在一个或者多个云提供者的不同区域中，则会带来重大的网络损耗。
+* 降低集群的隔离：当联邦控制平台中存在一个故障时，会影响所有的集群。把联邦控制平台的逻辑降到最小可以缓解这个问题。 无论何时，它都是kubernetes集群里控制平台的代表。在安全性和避免多集群中断方面，设计和应用也会犯错误。
+* 完整性：联邦项目相对较新，还不是很成熟。不是所有资源都可用，且很多资源才刚刚开始。[Issue 38893](https://github.com/kubernetes/kubernetes/issues/38893) 列举了一些团队正忙于解决的系统已知问题。
+
+### 混合云的能力
+
+Kubernetes集群里的联邦包括运行在不同云供应商上的集群（例如，谷歌云、亚马逊），和本地部署的集群（例如，OpenStack）。只需在适当的云供应商和/或位置创建所需的所有集群，并将每个集群的API endpoint和凭据注册到您的联邦API服务中（详情参考[联邦管理指南](/docs/admin/federation/)）。
+
+在此之后，您的[API资源](#api-resources)就可以跨越不同的集群和云供应商。
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/concepts/cluster-administration/federation.md]: api-resources
此处也同上处

File: cn/docs/concepts/cluster-administration/federation.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  - 为每个集群的固定资源花费降低开销，例如，使用apiserver的虚拟机（但是在全体集群开销中，中小型集群的开销占比要小的多）。
+
+多集群的原因包括：
+
+  - 严格的安全性策略要求隔离一类工作与另一类工作（但是，请参见下面的集群分割）。
+  - 测试集群或其他集群软件直至最优的新Kubernetes版本发布。
+
+## 选择合适的集群数
+
+因为仅偶尔回顾，Kubernetes集群数量选择也许是一个相对静止的选择。相比之下，一个集群中的节点数和一个服务中的pods数可能会根据负载和增长频繁变化。
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/concepts/cluster-administration/federation.md]: “因为仅偶尔回顾”？
理解不准确。

File: cn/docs/concepts/cluster-administration/federation.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+{% capture body %}
+## 为何使用联邦
+
+联邦可以使管理多个集群简单化。它提供了两个主要构件模块：
+
+  * 跨集群同步资源：联邦能够让资源在多个集群中同步。例如，你可以确保在多个集群中存在同样的部署。
+  * 跨集群发现：联邦能够在所有集群的后端自动配置DNS服务和负载均衡。例如，通过多个集群的后端，你可以确保全局的VIP或DNS记录可用。
+
+一些联邦使能的其他用例：
```
Comment (tengqm) [inline cn/docs/concepts/cluster-administration/federation.md]: 联邦技术的其他应用场景：

File: cn/docs/concepts/cluster-administration/federation.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+* 故障隔离：对于故障隔离，也许有多个小的集群比有一个大的集群要更好一些（例如：一个云供应商的不同可用域里有多个集群）。详细信息请参阅[多集群指南](/docs/admin/multi-cluster)。
+* 可伸缩性：对于单个kubernetes集群是有伸缩性限制的（但对于大多数用户来说并非如此。更多细节参考[Kubernetes扩展和性能目标](https://git.k8s.io/community/sig-scalability/goals.md)）。
+* [混合云](#hybrid-cloud-capabilities)：可以有多个集群，它们分别拥有不同的云供应商或者本地数据中心。
+
+### 注意事项
+
+虽然联邦有很多吸引人的用例，但这里还是有一些需要关注的事项：
+
+* 增加网络的带宽和损耗：联邦控制平台会监控所有的集群，来确保集群的当前状态与预期一致。那么当这些集群运行在一个或者多个云提供者的不同区域中，则会带来重大的网络损耗。
+* 降低集群的隔离：当联邦控制平台中存在一个故障时，会影响所有的集群。把联邦控制平台的逻辑降到最小可以缓解这个问题。 无论何时，它都是kubernetes集群里控制平台的代表。在安全性和避免多集群中断方面，设计和应用也会犯错误。
```
Comment (tengqm) [inline cn/docs/concepts/cluster-administration/federation.md]: 控制平台？

Comment (tengqm) [reply cn/docs/concepts/cluster-administration/federation.md]: ”在安全性和避免多集群中断方面，设计和应用也会犯错误。“

不理解

File: cn/docs/concepts/cluster-administration/federation.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+### 混合云的能力
+
+Kubernetes集群里的联邦包括运行在不同云供应商上的集群（例如，谷歌云、亚马逊），和本地部署的集群（例如，OpenStack）。只需在适当的云供应商和/或位置创建所需的所有集群，并将每个集群的API endpoint和凭据注册到您的联邦API服务中（详情参考[联邦管理指南](/docs/admin/federation/)）。
+
+在此之后，您的[API资源](#api-resources)就可以跨越不同的集群和云供应商。
+
+## 建立联邦
+
+若要能联合多个集群，首先需要建立一个联邦控制平台。参照[安装指南](/docs/tutorials/federation/set-up-cluster-federation-kubefed/) 建立联邦控制平台。
```
Comment (tengqm) [inline cn/docs/concepts/cluster-administration/federation.md]: control plane一般译成控制面

File: cn/docs/concepts/cluster-administration/federation.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+[API参考文档](/docs/reference/federation/)列举了联邦API服务支持的所有资源。
+
+## 级联删除
+
+Kubernetes1.6版本支持联邦资源级联删除。使用级联删除，即当删除联邦控制平台的一个资源时，也删除了所有底层集群中的相应资源。
+
+当使用REST API时，级联删除功能不是默认开启的。若使用REST API从联邦控制平台删除一个资源时，要开启级联删除功能，即需配置选项 `DeleteOptions.orphanDependents=false`。使用`kubectl delete`使级联删除功能默认开启。使用`kubectl delete --cascade=false`禁用级联删除功能。
+
+注意：Kubernetes1.5版本支持联邦资源子集的级联删除。
```
Comment (tengqm) [inline cn/docs/concepts/cluster-administration/federation.md]: 支持->开始支持

File: cn/docs/concepts/cluster-administration/federation.md [hunk 8]
```diff
@@ -94,7 +94,7 @@ Kubernetes1.6版本支持联邦资源级联删除。使用级联删除，即当
 
 ## 选择合适的集群数
 
-因为仅偶尔回顾，Kubernetes集群数量选择也许是一个相对静止的选择。相比之下，一个集群中的节点数和一个服务中的pods数可能会根据负载和增长频繁变化。
+Kubernetes集群数量选择也许是一个相对静止的选择，因为对其重新审核的情况很少。相比之下，一个集群中的节点数和一个服务中的pods数可能会根据负载和增长频繁变化。
```
Comment (tengqm) [inline cn/docs/concepts/cluster-administration/federation.md]: 静止 -> 稳定

**Comment by zhangxiaoyu-zidif** [comment]
@tengqm 
how about this PR now？

**Pull request merged**